client/src/reducers/annotations.ts

```diff
@@ -0,0 +1,82 @@
+import type { AnyAction } from "redux";
```

Comment:
New reducer tracks UI-only state for category editing dialogs (e.g., whether rename modal is open). These flags prevent autosave from racing against the modal flow.